deployment/collector.go

```diff
@@ -205,3 +212,47 @@ func (c Collector) zipDir(path string) (string, error) {
 
 	return targetPth, nil
 }
+
+func (c Collector) mergeZipPairs(deployableItems []DeployableItem) []DeployableItem {
+	var mergedDeployableItems []DeployableItem
+	pipelineDirs := map[string]DeployableItem{}
+	zipBuildArtifacts := map[string]DeployableItem{}
+
+	for _, item := range deployableItems {
+		if item.IntermediateFileMeta != nil {
+			if item.IntermediateFileMeta.IsDir {
+				pipelineDirs[item.Path] = item
+			}
+			mergedDeployableItems = append(mergedDeployableItems, item)
+			continue
+		}
+
+		if filepath.Ext(item.Path) == ".zip" {
+			zipBuildArtifacts[item.Path] = item
+		} else {
+			mergedDeployableItems = append(mergedDeployableItems, item)
+		}
+	}
+
+	// At this point mergedDeployableItems contains all Pipeline Files and none ZIP Build Artifacts.
```

Comment:
Minor nitpick, I would change the `none` to `no` in this sentence. It sounds better with `no` to me. 😄

Reply:
Updated in https://github.com/bitrise-steplib/steps-deploy-to-bitrise-io/pull/168/commits/ffd454ac2b316395b840ded96e1f061b3ce6c05c